packages/core/src/VNodes/ContainerNode.ts

```diff
@@ -182,26 +51,24 @@ export class ContainerNode extends AbstractNode {
      * See {@link AbstractNode.insertBefore}.
      */
     insertBefore(node: VNode, reference: VNode): void {
-        const index = this.childVNodes.indexOf(reference);
-        if (index < 0) {
-            throw new ChildError(this, node);
-        }
-        this._insertAtIndex(node, index);
+        this._checkChildError(reference);
```

Comment:
Why should it throw an error when inserting before `!parentVNode.tangible` ?

Reply:
Also, I would add a better error message to indicate the source of the Error.

Reply:
is not a child because by default we don't take care of the not tangible nodes

Reply:
and the why, because we can only insert the node if the child have this parent...